test/template_resolver_tests.js

```diff
@@ -54,6 +54,7 @@ describe('Template Resolver', () => {
       it(`resolves template script for ${t.url} correctly`, async () => {
         const mockReq = {
           url: t.url,
+          get: () => 'localhost',
```

Comment:
I would only return `localhost` for `host`, eg:

```suggestion
          get: (key) => (key === 'host' ? 'localhost' : ''),
```